Graphics/Slicer/Machine/Infill.hs

```diff
@@ -82,18 +80,19 @@ infillLineSegInside contour childContours line
                   saneIntersection _ (Left NoIntersection)         _ = Nothing
                   saneIntersection _ (Right PParallel)             _ = Nothing
                   saneIntersection _ (Right PAntiParallel)         _ = Nothing
-                  saneIntersection _                          (Left (HitStartPointL2 point)) (Left (HitEndPointL2   _)) = Just point
-                  saneIntersection (Left (HitStartPointL2 _)) (Left (HitEndPointL2   _    )) _                          = Nothing
-                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Left (HitStartPointL2 _)) = Nothing
-                  saneIntersection (Left (HitEndPointL2   _)) (Left (HitStartPointL2 point)) _                          = Just point
-                  saneIntersection (Right PColinear)          (Left (HitStartPointL2 _    )) _                          = Nothing
-                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Right PColinear)          = Nothing
+                  saneIntersection _                          (Left (HitStartPoint _ point)) (Left (HitEndPoint   _ _)) = Just point
```

Comment:
I don't understand why the cases in this function are what they are.  Why is `saneIntersection _ HitEndPoint{} HitStartPoint{}` nothing? Some comments or helper functions would really help aid in understanding of the logic here.

Reply:
added comments.